### ISSUE TITLE
[JOSS REVIEW]: Add automated tests

### ISSUE THREAD

**Issue by finsberg** [CastillonMiguel/phasefieldx#3]
The documentation for the library looks really nice and it seems to me that all the examples are actually run when you build the documentation on readthedocs. In this way you could argue that the examples in the documentation actually serves as tests.
However, I think it would be good to have some automated tests that can run whenever someone pushes to the repo or open a pull request so that potential contributors (and yourself) can be notified if you break some functionality. I am not so familiar with readthedocs, and it might be that you will get notified if building the docs fails?

Since you already have the examples in place, you could just create a github workflow that runs the examples in CI. This way you can also more easily upgrade to a new version by just trying a new docker image. 

If the examples also run in parallel, then if would also be good to add a test for that. 

I did something similar here https://github.com/RangamaniLabUCSD/smart/blob/development/.github/workflows/run_examples_mpi.yml (allthough this is using Legacy FEniCS). 

This issue is related to the review here: https://github.com/openjournals/joss-reviews/issues/7307

**Comment by CastillonMiguel** [comment]
Yes, I get notified if the documentation build fails. When I push changes, ReadTheDocs builds all the documentation and runs all the examples. If something is wrong, the documentation build fails, and I receive an error.  However, most of the current examples have a high computational cost, so I preload the results directly, meaning they don’t serve as tests.

The best approach might be to add some new, simple automated tests to help catch potential errors. I will implement these new tests. Thanks!

**Comment by CastillonMiguel** [comment]
I have added a basic test framework. All tests are located in a new folder called test/, and they are run using pytest. I am now checking some of the models with simple test cases that allow for comparison of the numerical solutions with the analytical ones. Additionally, I have included tests related to general utilities. Furthermore, I have implemented a GitHub workflow that automatically runs all tests whenever a push or pull request is made. 
https://github.com/CastillonMiguel/phasefieldx/commit/4eeee4014a533905ec7b7e9908b0a0607f9b1cb1
